File: src/Model/Statement/CompoundStatement.java
package Model.Statement;

import Model.ProgramState;
import DataTypes.MyIStack;

/**
 * Created by vladc on 12.10.2016.
 */
public class CompoundStatement implements IStatement {
    private IStatement first;
    private IStatement second;

    public CompoundStatement(IStatement first, IStatement second) {
        this.first = first;
        this.second = second;
    }

    @Override
    public String toString() {
        return "(" + first.toString() + " ; " + second.toString() + ")";
    }

    public ProgramState execute(ProgramState state) {
        MyIStack<IStatement> st = state.getExeStack();
        st.push(this.second);
        st.push(this.first);
        return state;
    }
}

File: Toy-Language-Interpreter/Model/Statement/CompoundStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Toy_Language_Interpreter.DataTypes;

namespace Toy_Language_Interpreter.Model.Statement
{
    class CompoundStatement : IStatement
    {
        private IStatement first;
        private IStatement second;
        public CompoundStatement(IStatement first, IStatement second)
        {
            this.first = first;
            this.second = second;
        }
        public ProgramState Execute(ProgramState state)
        {
            MyIStack<IStatement> st = state.GetExeStack();
            st.Push(this.second);
            st.Push(this.first);
            return state;
        }

        public override string ToString()
        {
            return "(" + first.ToString() + " ; " + second.ToString() + ")";
        }
    }
}
